codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/OperationGenerator.java

```diff
@@ -166,18 +171,89 @@ private void constructHandler() {
      * Adds middleware to the operation middleware stack.
      */
     private void populateOperationMiddlewareStack() {
+
+        // Populate middleware's from runtime client plugins
         runtimeClientPlugins.forEach(runtimeClientPlugin -> {
             if (!runtimeClientPlugin.matchesService(model, service)
                     && !runtimeClientPlugin.matchesOperation(model, service, operation)) {
                 return;
             }
 
-            if (runtimeClientPlugin.buildMiddlewareStack().isPresent()) {
-                runtimeClientPlugin.buildMiddlewareStack().get().applyMiddleware(
-                        writer, service, operation, protocolGenerator, "stack"
-                );
+            if (runtimeClientPlugin.registerMiddleware().isPresent()) {
```

Comment:
Not super important, but in cases like this making these if statements guard clauses can improve readability by reducing indentation. i.e. `if (!condition) { break; }`